git-sync/Dockerfile

```diff
@@ -28,18 +28,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # https://github.com/kubernetes/git-sync/releases
-FROM registry.k8s.io/git-sync/git-sync:v4.4.1 AS upstream
-
-FROM quay.io/astronomer/ap-base:3.21.3-5
+FROM quay.io/astronomer/ap-base:3.21.3-6
 LABEL maintainer="Astronomer <humans@astronomer.io>"
-COPY --from=upstream /git-sync /usr/local/bin/
+
 USER root
-RUN apk update && apk upgrade --available && apk add --no-cache git openssh-client
+
+# Install git and openssh-client (Go 1.25.1 is already installed)
+RUN apk update && apk upgrade --available && \
+    apk add --no-cache openssh-client git
+
+# Build git-sync from source with Go 1.25.1
+RUN git clone https://github.com/kubernetes/git-sync.git /tmp/git-sync && \
+    cd /tmp/git-sync && \
+    git checkout v4.4.1 && \
+    CGO_ENABLED=0 go build -o /usr/local/bin/git-sync . && \
+    cd / && rm -rf /tmp/git-sync
```

Comment:
Does chainguard have a build of https://github.com/kubernetes/git-sync ? If not maybe they should, and we can ask them to build it.